Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift

```diff
@@ -0,0 +1,119 @@
+import UIKit
+import Checkout
+
+protocol ExpiryDateViewDelegate: AnyObject {
+    func update(expiryDate: ExpiryDate)
+}
+
+final class ExpiryDateView: UIView {
+    weak var delegate: ExpiryDateViewDelegate?
+
+    private(set) var style: CellTextFieldStyle?
+
+    private lazy var cardValidator: CardValidator = {
+        CardValidator(environment: .production)
+    }()
+
+    private lazy var expiryDateView: InputView = {
+        let view = InputView().disabledAutoresizingIntoConstraints()
+        view.delegate = self
+        return view
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        // setup expiry DateView
+        addSubview(expiryDateView)
+        expiryDateView.setupConstraintEqualTo(view: self)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func update(style: CellTextFieldStyle?) {
+        self.style = style
+        expiryDateView.update(style: style)
+    }
+
+    private func updateErrorView(isHidden: Bool, text: String?){
+        style?.error?.isHidden = isHidden
+        style?.textfield.text = text ?? ""
+        expiryDateView.update(style: style)
+    }
+
+    @discardableResult
+    func isValidExpiryDate(text: String?) -> Bool {
```

Comment:
For a method that is appearing at call site to just validate something, there are quite a few side effects, mutating internal structure. Could this logic be improved at all?

My first thought would be to follow an idea of:
- rename function to be more in line of `updateExpiryDate(to newDate: String)`
- exposed property (`hasValidExpiryDate`), which is updated by the function

So at call site, instead of calling `isValidExpiryDate(text: "ABC")`, the consumer would have to use:
```
view.updateExpiryDate(to: "ABC")
let isStringValidDate = view.hasValidExpiryDate
```